src/uiTest/resources/ideprobe.conf

```diff
@@ -22,6 +22,17 @@ probe {
     }
   }
 
+  intellij {
+    plugins = [
+      { uri = ${IDEPROBE_INTELLIJ_PLUGIN_URI} }
+    ]
+
+    # We're cheating here a bit since IDEPROBE_INTELLIJ_VERSION_BUILD might be either a build number or a release number,
+    # while we're always treating it as a build number.
+    # Still, as of ide-probe 0.42.0, even when release number like `2020.3` is passed as `version.build`, UI tests work just fine.
+    version.build = ${IDEPROBE_INTELLIJ_VERSION_BUILD}
```

Comment:
It works for ide-probe `0.42.0` but... After latest changes from https://github.com/VirtusLab/ide-probe/pull/252 - which should be released as `0.43.0` - it probably will not work well. The previous solution on `git-machete-intellij-plugin` used:
`fixtureFromConfig("ideprobe.conf").withVersion(intelliJVersion)`
which overrides the `IntelliJVersion` **after parsing configuration**. And this previous solution created an object of type `IntellijVersion` with `build` field specified correctly and `release` was a `None`. So, `ide-probe` used only the `build` parameter which worked just fine. New solution does not use that _trick_.

In effect, newer versions of`ide-probe` will get `version.build` from git-machete `ideprobe.conf` and `version.release` from the default ide-probe `reference.conf`. And that would result in issues like downloading not the version we specified here. If we want to go only with `ideprobe.conf` without the _withVersion_ tricks - we have to specify both `version.build` and `version.release` - so that they would match properly.

Anyway - did you test it against a snapshot version of `ide-probe` with changes from https://github.com/VirtusLab/ide-probe/pull/252 included?

Reply:
These change will be OK if we want to stick with `0.42.0` version of ide-probe. But for newer releases - this will not work well. The problem is also related to removing the default values from scala code of the ide-probe, which is connected with the way our configs are passed here: https://github.com/VirtusLab/ide-probe/blob/master/core/driver/sources/src/main/scala/org/virtuslab/ideprobe/ide/intellij/IntelliJProvider.scala#L184

Reply:
Alternatively - this changes could work well if we make a small change in the default `reference.conf` in ide-probe:
change from:
```
intellij {
    plugins = []
    version {
      build = "212.5080.55"
      release = "2021.2.1"
    }
```

to

```
intellij {
    plugins = []
    version {
      build = "212.5080.55"
      # release = ""
    }
```
so that `release` will be `None` by default

Reply:
Whooooops good spot, I didn't take it into account ☹️

Huh let's go deeper for a second... why is even this **double representation** needed by ide-probe? 🤔 I mean, if someone specifies `build` and `release` which point to different versions... what's gonna be the effect? Lemme open an issue for that in ide-probe...

Reply:
https://github.com/VirtusLab/ide-probe/issues/265, let's consider it a blocker for this PR then... changing to draft

Reply:
Ha! actually, as for now I think I have a simple workaround that would work even on 0.43.0+:

```
probe.intellij.version.release = null
```

See the newly-pushed commit. Un-drafting the PR

Reply:
Lemme still check it against a snapshot version of ide-probe, however...

Reply:
Looks like it works fine now!